packages/cluster/package.json

```
@@ -9,9 +9,14 @@
   "license": "MIT",
   "exports": {
     ".": {
-      "types": "./index.d.ts",
-      "import": "./build.mjs",
-      "require": "./build.js"
+      "import": {
+        ".": "./build.mjs",
+        "types": "./index.d.mts"
+      },
+      "require": {
+        ".": "./build.js",
+        "types": "./index.d.ts"
+      }
```

For all modules, please make these changes:

* TS says its requires "types" to be first
* Rename "." -> "default" & order it after "types"
* Rename `index.d.ts` as `index.d.mts`
* Add new  `index.d.ts` which is the CommonJS export DTS file (to match "exports" description)

```suggestion
      "import": {
        "types": "./index.d.mts",
        "default": "./build.mjs"
      },
      "require": {
        "types": "./index.d.ts",
        "default": "./build.js",
      }
```

---

> Are you suggesting to use `export = ` 

Yes, to lean into the CommonJS TS definitions, rather than relying on tsc's implicit conversion

> Any chance you'd consider dropping the CJS build? 😉

No

---

Ok, I just pushed that change. I don't have a lot of experience with that style. I spent like an hour tying to find documentation and examples for it and it's not used much as far as I could see. Hopefully I did it correctly. If I'm way off base I think we could just revert the last commit. I feel it's a bit out of scope to update the types for CJS users and we could probably leave them as is in this PR. I was really just trying to get the ESM types updated, which I'm a lot more familiar with.